caffe2/utils/Macros.h

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+/**
+ * Disallow the copy and assignment constructors of a class
+ */
+#define DISALLOW_COPY_AND_ASSIGN(Class)         \
```

Comment:
Existing code here:

https://github.com/pytorch/pytorch/blob/master/caffe2/core/common.h#L65-L72

consider including core/common.h or move core/common.h to utils here.